packages/admin/components/composed/collection/CollectionAddDrawer/CollectionAddDrawer.tsx

```diff
@@ -18,33 +19,57 @@ export default function CollectionAddDrawer({
   const { drawerSlug } = params;
   const { t } = useTranslation();
 
-  return (
+  function renderDrawer(data: any) {
+    return (
+      <Drawer
+        dialog={dialog}
+        hideOnClickOutside={false}
+        label={t("actions.add.collection")}
+        header={t("actions.add.collection_header")}
+      >
+        {data && (
+          <CollectionAddForm
+            data={data}
+            onSuccess={dialog.hide}
+            onCancel={dialog.hide}
+            parentId={data?.collection?.id ?? data?.community?.id}
+          />
+        )}
+      </Drawer>
+    );
+  }
+
+  return drawerSlug ? (
     <QueryWrapper<Query>
       query={query}
-      initialVariables={{ parentSlug: drawerSlug }}
+      initialVariables={{ parentSlug: drawerSlug, schemaKind: "COLLECTION" }}
+    >
+      {({ data }) => renderDrawer(data)}
+    </QueryWrapper>
+  ) : (
+    <QueryWrapper<CollectionAddDrawerRootQuery>
+      query={rootQuery}
+      initialVariables={{ schemaKind: "COLLECTION" }}
     >
-      {({ data }) => (
-        <Drawer
-          dialog={dialog}
-          hideOnClickOutside={false}
-          label={t("actions.add.collection")}
-          header={t("actions.add.collection_header")}
-        >
-          {data && (
-            <CollectionAddForm
-              data={data}
-              onSuccess={dialog.hide}
-              onCancel={dialog.hide}
-            />
-          )}
-        </Drawer>
-      )}
+      {({ data }) => renderDrawer(data)}
     </QueryWrapper>
   );
 }
 
 const query = graphql`
-  query CollectionAddDrawerQuery($parentSlug: Slug!) {
+  query CollectionAddDrawerQuery($parentSlug: Slug!, $schemaKind: SchemaKind!) {
+    collection(slug: $parentSlug) {
+      id
+    }
+    community(slug: $parentSlug) {
+      id
+    }
+    ...CollectionAddFormFragment
+  }
+`;
+
+const rootQuery = graphql`
+  query CollectionAddDrawerRootQuery($schemaKind: SchemaKind!) {
```

Comment:
Both @1aurend  and I felt it was really silly to have to forms for creating a Collection on the root page and interior collection pages, but the implementation is a little funky. The parentSlug is a required field if you are running the collection or community query, so we can't pass an empty value to parentSlug. This is true even if you use the graphql `includes` directive to conditionally fetch this information. 

So what I have here is two different Query wrappers, one with the parentSlug and one without. This means we can use the same CollectionAddForm for both, and only render the Communities list if a parentId is not passed.